### DISCUSSION TITLE
taxonomy url results in 404

### DISCUSSION THREAD

**Discussion by torbentschechne** [statamic/cms#4692] [Troubleshooting] [answered]
Hello,

I am migrating my site from v2 to v3 and in v2 I have used something like this:

```
{{ taxonomy:tags }}
    <div class="w-1/3 px-2">
        <a href="{{ url }}">{{ title }}</a>
    </div>
{{ /taxonomy:tags }}
```

This results in a link to e.g `domain.com/tags/newsletter` but when I click on it, it results in a 404. The `content/taxonomies/tags.yaml` is set to: `route: '/tags/{slug}'`

What am I missing here?

**Comment by jasonvarga** [comment, marked as answer]
Taxonomies work differently in v3. You don't set the `route`. It's automatically going to be at `/{taxonomy}/{term}` (e.g. /tags/newsletter) but only if the view exists. Make sure you have a `resources/views/tags/show.antlers.html` view.

https://statamic.dev/taxonomies#routing

**Reply by torbentschechne** [reply]
Thanks. That's it.

**Comment by breizhwave** [comment]
funny in my case it only worked when I put the tags folder under blog.. any idea ? 
![image](https://user-images.githubusercontent.com/16898162/160297698-ba8d8ccb-8830-470c-bcf4-ae990fefb64e.png)

**Reply by jasonvarga** [reply]
Putting them in the subdirectory should enable the /blog/{taxonomy} route.

**Reply by VitoScarangella** [reply]
Hi everyone, i'm having the same problem with taxonomies routing and i cannot figure out how it works, even because i have to handle multi-site.

I have views in this order:
![image](https://user-images.githubusercontent.com/29036906/171212222-abf44aa7-fce2-44fa-bd47-c629c6a586bd.png)

As @jasonvarga said in the last comment, putting the view in the subdirectory enables the domain.com/blog/categories/{slug} and it works.

The problem is that domain.com/categories/{slug} works too. This is duplicate content for SEO purpose and is not good.
The same for tags taxonomy.

Other problem for multisite: 

domain.com/en/categories/{slug} result in 404 (And this is right, but why that route is available in first site?).

domain.com/en/blog/categories/{slug} works fine.

_I already enabled the taxonomy for both first and second site in control panel_

If i understand right from docs, domain.com/categories/{slug} should be available only if categories folder with show.antlers.html and index.antlers.html inside would be available in the views folder directly, and not in blog subdirectory.

Other thing that maybe can help you to understand the problem:
I'm looping through taxonomies terms in this way:
![image](https://user-images.githubusercontent.com/29036906/171215304-324079c3-dcc2-4a34-86ef-5b7abeac2cf1.png)
**the {url} in this case is /categories/{slug}**

If instead i loop in the articles and i loop in specific article's categories:
![image](https://user-images.githubusercontent.com/29036906/171216236-61b0ab4c-a328-4894-b5a9-870c9e0e9b74.png)
**the {url} in this case is /blog/categories/{slug}**

Any idea of why this happen? Thanks in advance to everyone who will answer to me, i'm losing hair already for this problem.
